src/main/java/com/marklogic/kafka/connect/source/JsonRowBatcherBuilder.java

```diff
@@ -0,0 +1,52 @@
+package com.marklogic.kafka.connect.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.RowBatchSuccessListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.marker.ContentHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import java.util.List;
+import java.util.Map;
+
+public class JsonRowBatcherBuilder extends LoggingObject implements RowBatcherBuilder<JsonNode> {
+    private final String topic;
+    private final DataMovementManager dataMovementManager;
+
+    JsonRowBatcherBuilder(DataMovementManager dataMovementManager, Map<String, Object> parsedConfig) {
+        this.dataMovementManager = dataMovementManager;
+        this.topic = (String) parsedConfig.get(MarkLogicSourceConfig.TOPIC);
+    }
+
+    public RowBatcher<JsonNode> newRowBatcher(List<SourceRecord> newSourceRecords) {
+        ContentHandle<JsonNode> contentHandle = new JacksonHandle().withFormat(Format.JSON).withMimetype("application/json");
+        RowBatcher<JsonNode> rowBatcher =  dataMovementManager.newRowBatcher(contentHandle);
+        rowBatcher.onSuccess(event -> onSuccessHandler(event, newSourceRecords));
+        return rowBatcher;
+    }
+
+    protected void onSuccessHandler(RowBatchSuccessListener.RowBatchResponseEvent<JsonNode> event, List<SourceRecord> newSourceRecords) {
+        JsonNode rows = event.getRowsDoc().get("rows");
+        logger.debug("JsonNode: \n" + rows.toPrettyString());
+
+        for (JsonNode row : rows) {
+// We may need to add a switch to include a key in the record depending on how the target topic is configured.
+// If the topic's cleanup policy is set to "compact", then a key is required to be included in the SourceRecord.
+//            String key = event.getJobBatchNumber() + "-" + rowNumber;
+
+            // Calling toString on the JsonNode, as when this is run in Confluent Platform, we get the following
+            // error: org.apache.kafka.connect.errors.DataException: Java class com.fasterxml.jackson.databind.node.ObjectNode does not have corresponding schema type.
+            try {
+                SourceRecord newRecord = new SourceRecord(null, null, topic, null, row.toString());
+                newSourceRecords.add(newRecord);
+            } catch (Exception ex) {
+                logger.error("Failed to create or add SourceRecord from result row; cause: " + ex.getMessage());
```

Comment:
Just do one logger.error call, can stuff everything together - having two log "events" for one error can make life difficult for users that e.g. send log messages to a tool like Grafana, where ideally an error is represented by a single log event (so the error doesn't get counted twice).